From 4caf72e60bb6fe7bf8fb0c3f1e2cafe4421ef9c1 Mon Sep 17 00:00:00 2001
From: Naseer Ahmed <naseer@codeaurora.org>
Date: Fri, 11 Oct 2013 10:43:48 -0400
Subject: [PATCH] msm: rotator: signal timeline when rotator prepare fails.

When rotator prepare fails, timeline is not signalled causing fence
timeout on the rotator release fence. This patch fixes the issue.

Change-Id: I6023393d768dbb1a0a9095bb1822ac504632ed64
Signed-off-by: Vishnuvardhan Prodduturi <vproddut@codeaurora.org>
Signed-off-by: Naseer Ahmed <naseer@codeaurora.org>
---
 drivers/char/msm_rotator.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/char/msm_rotator.c b/drivers/char/msm_rotator.c
index 7ddeb74..aaf42d1 100644
--- a/drivers/char/msm_rotator.c
+++ b/drivers/char/msm_rotator.c
@@ -1980,6 +1980,7 @@ static int msm_rotator_rotate_prepare(
 		fput_light(srcp0_file, ps0_need);
 	else
 		put_img(srcp0_file, srcp0_ihdl, ROTATOR_SRC_DOMAIN, 0);
+	msm_rotator_signal_timeline_done(s);
 	dev_dbg(msm_rotator_dev->device, "%s() returning rc = %d\n",
 		__func__, rc);
 	mutex_unlock(&msm_rotator_dev->rotator_lock);
